astropy/config/paths.py

```diff
@@ -104,29 +108,7 @@ def get_cache_dir_path(rootname: str = "astropy") -> Path:
         The absolute path to the cache directory.
 
     """
-    # symlink will be set to this if the directory is created
-    linkto = None
-
-    # If using set_temp_cache, that overrides all
-    if set_temp_cache._temp_path is not None:
-        xch = set_temp_cache._temp_path
-        cache_path = xch / rootname
-        if not cache_path.is_file():
-            cache_path.mkdir(exist_ok=True)
-        return cache_path.resolve()
-
-    # first look for XDG_CACHE_HOME
-    if (
-        (xdg_config_home := os.getenv("XDG_CONFIG_HOME")) is not None
```

Comment:
it should have been XDG_CACHE_HOME here I think, so currently the code is using only XDG_CONFIG_HOME I think, not XDG_CACHE_HOME.

Reply:
Indeed. For posterity this is being addressed in https://github.com/astropy/astropy/pull/17514